apps/postgresql.go

```diff
@@ -0,0 +1,219 @@
+}
+
+func (LoggingProcessorPostgresql) Type() string {
+	return "postgresql_general"
+}
+
+func (p LoggingProcessorPostgresql) Components(tag string, uid string) []fluentbit.Component {
+	c := confgenerator.LoggingProcessorParseMultilineRegex{
+		LoggingProcessorParseRegexComplex: confgenerator.LoggingProcessorParseRegexComplex{
+			Parsers: []confgenerator.RegexParser{
+				{
+					// Limited logging documentation: https://www.postgresql.org/docs/10/runtime-config-logging.html
+					// Sample line: 2022-01-12 20:57:58.378 UTC [26241] LOG:  starting PostgreSQL 14.1 (Debian 14.1-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
+					// Sample line: 2022-01-12 20:59:25.169 UTC [27445] postgres@postgres FATAL:  Peer authentication failed for user "postgres"
+					// Sample line: 2022-01-12 21:49:13.989 UTC [27836] postgres@postgres LOG:  duration: 1.074 ms  statement: select *
+					//    from pg_database
+					//    where 1=1;
+					Regex: `^(?<time>\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3,} \w+)\s*\[(?<tid>\d+)\](?:\s+(?<role>\S*)@(?<user>\S*))?\s*(?<level>\w+):\s+(?<message>[\s\S]*)`,
+					Parser: confgenerator.ParserShared{
+						TimeKey:    "time",
+						TimeFormat: "%Y-%m-%d %H:%M:%S.%L %z",
+						Types: map[string]string{
+							"tid": "integer",
+						},
+					},
+				},
+			},
+		},
+		Rules: []confgenerator.MultilineRule{
+			{
+				StateName: "start_state",
+				NextState: "cont",
+				Regex:     `\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3,} \w+`,
+			},
+			{
+				StateName: "cont",
+				NextState: "cont",
+				Regex:     `^(?!\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}.\d{3,} \w+)`,
+			},
+		},
+	}.Components(tag, uid)
+
+	// https://www.postgresql.org/docs/10/runtime-config-logging.html#RUNTIME-CONFIG-SEVERITY-LEVELS
+	c = append(c,
+		fluentbit.TranslationComponents(tag, "level", "logging.googleapis.com/severity",
+			[]struct{ SrcVal, DestVal string }{
+				{"DEBUG1", "DEBUG"},
+				{"DEBUG2", "DEBUG"},
+				{"DEBUG3", "DEBUG"},
+				{"DEBUG4", "DEBUG"},
+				{"DEBUG5", "DEBUG"},
+				{"DETAIL", "DEBUG"},
+				{"STATEMENT", "DEBUG"},
+				{"INFO", "INFO"},
+				{"LOG", "INFO"},
+				{"NOTICE", "INFO"},
+				{"ERROR", "ERROR"},
+				{"WARNING", "WARNING"},
+				{"FATAL", "CRITICAL"},
+				{"PANIC", "CRITICAL"},
+			},
+		)...,
+	)
+
+	return c
+}
+
+type LoggingReceiverPostgresql struct {
+	LoggingProcessorPostgresql              `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (r LoggingReceiverPostgresql) Components(tag string) []fluentbit.Component {
+	if len(r.IncludePaths) == 0 {
+		r.IncludePaths = []string{
+			// Default log paths for Debain / Ubuntu
+			"/var/log/postgresql/postgresql*.log",
+			// Default log paths for SLES
+			"/var/lib/pgsql/data/log/postgresql*.log",
+			// Default log paths for CentOS / RHEL
+			"/var/lib/pgsql/*/data/log/postgresql*.log",
+			// "/var/lib/pgsql/14/data/log/postgresql*.log",
```

Comment:
Are these comments removable? Or is there something implicit that we are trying to point out?

Reply:
Those can be removed, I was attempting to lay out the expected paths signified by the asterisk in the default paths for CentOS/RHEL, but can remove.

Reply:
SG